utils.go

```diff
@@ -57,3 +58,30 @@ func parseImageDestination(c *cli.Context, name string) (types.ImageDestination,
 	}
 	return nil, fmt.Errorf("Unrecognized image reference %s", name)
 }
+
+func parseDockerImage(img, certPath string, tlsVerify bool) (types.Image, error) {
+	s, err := newDockerImageSource(img, certPath, tlsVerify)
+	if err != nil {
+		return nil, err
+	}
+	return &dockerImage{src: s}, nil
+}
+
+// parseDockerImageName converts a string into a reference and tag value.
+func parseDockerImageName(img string) (reference.Named, string, error) {
```

Comment:
Similarly, this would probably be better in `docker_client.go` - it is an internal helper shared between the `newDockerImageSource` and `NewDockerImageDestination` constructors, i.e. any user of `dockerImageSource` will need this function as well; so pulling `docker_*.go` into a separate subpackage would drag in the top-level package for `utils.go`.